zarr/tests/test_core.py

```diff
@@ -852,9 +855,10 @@ def test_structured_array(self):
                 assert_array_equal(a['bar'], z['bar'])
                 assert_array_equal(a['baz'], z['baz'])
 
-            with pytest.raises(ValueError):
-                # dodgy fill value
-                self.create_array(shape=a.shape, chunks=2, dtype=a.dtype, fill_value=42)
+            # this does not raise with numpy 1.14
+            # with pytest.raises(ValueError):
+            #     # dodgy fill value
+            #     self.create_array(shape=a.shape, chunks=2, dtype=a.dtype, fill_value=42)
```

Comment:
Maybe we can add a version check with this test? Also do we know why this doesn't raise with NumPy 1.14?

Reply:
Numpy 1.14 is happy with a fill value like 42 for a structured array, it actually fills every field with the value, whereas numpy 1.13 barfs. I thought about adding a version check but then thought maybe that was overcomplicating, happy to add one if you think it's worth it.

Reply:
Interesting. No strong opinions from me.

That said, if we think this test is no longer worthwhile, we should delete it instead of commenting it. Commented code has a way of sticking around unaddressed. It's much easier to add a new PR marked WIP readding this code and it has a better chance of being addressed that way (if needed).